auxiliary/crash-handler/CMakeLists.txt

```diff
@@ -0,0 +1,95 @@
+project(chatterino-crash-handler 
+        VERSION 0.1.0
+        LANGUAGES CXX
+)
+set(PROJECT_LIB "${PROJECT_NAME}-lib")
+
+set(LIB_SOURCE_FILES
+    commandline.hpp
+    commandline.cpp
+    recovery.hpp
+    recovery.cpp
+)
+
+set(EXE_SOURCE_FILES
+    main.cpp
+)
+
+if(WIN32)
+    list(APPEND LIB_SOURCE_FILES
+        win_support.hpp
+        win_support.cpp
+    )
+endif()
+
+add_library(${PROJECT_LIB} OBJECT ${LIB_SOURCE_FILES})
+target_link_libraries(${PROJECT_LIB}
+    PUBLIC
+        crashpad_handler_lib
+        crashpad_tools
+        MagicEnum
+)
+set_target_properties(${PROJECT_LIB}
+    PROPERTIES
+        CXX_STANDARD 23
+        CXX_STANDARD_REQUIRED On
+)
+
+add_executable(${PROJECT_NAME} WIN32 ${EXE_SOURCE_FILES})
+target_link_libraries(${PROJECT_NAME} PRIVATE ${PROJECT_LIB})
+set_target_properties(${PROJECT_NAME}
+    PROPERTIES
+        RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/bin/crashpad"
+        OUTPUT_NAME "crashpad_handler"
+        CXX_STANDARD 23
```

Comment:
C++23 seems extreme when we're barely able to use C++20 - what C++23 features are we using?

Reply:
[`std::ranges::fold_left`](https://en.cppreference.com/w/cpp/algorithm/ranges/fold_left).